drivers/lora/sx12xx_common.c

```diff
@@ -90,15 +94,42 @@ static bool modem_release(struct sx12xx_data *data)
 static void sx12xx_ev_rx_done(uint8_t *payload, uint16_t size, int16_t rssi,
 			      int8_t snr)
 {
-	dev_data.rx_buf = payload;
-	dev_data.rx_len = size;
-	dev_data.rssi = rssi;
-	dev_data.snr = snr;
-
-	modem_release(&dev_data);
+	struct k_poll_signal *sig = dev_data.operation_done;
```

Comment:
I think there is a corner case where you can incorrectly dereference a dangling pointer here. If you are very unlucky, you could potentially have an RX timeout followed by an immediate call to `sx12xx_lora_recv`. If this callback has been pending for some reason (e.g., preemption or lower priority work item), I think this could be called with `operation_done` in state 1. In theory, it seems like you could have a preemption betwen reading the `operation_done` pointer and pre-releasing the modem.

Reply:
As the comment below explains, I don't think this is possible.

`sig` is only used if the `atomic_cas` succeeds, and if that succeeds we can be sure the calling context is still in `lora_recv`.